content/200-orm/200-prisma-client/500-deployment/301-edge/485-deploy-to-vercel.mdx

```diff
@@ -0,0 +1,458 @@
+---
+title: 'Deploy to Vercel Edge Functions & Middleware'
+navTitle: 'Deploy to Vercel'
+metaTitle: 'Deploy to Vercel Edge Functions'
+metaDescription: 'Learn the things you need to know in order to deploy an Edge function that uses Prisma Client for talking to a database.'
+tocDepth: 3
+preview: true
+---
+
+<TopBlock>
+
+This page covers everything you need to know in order to deploy an edge function to [Vercel Edge Functions](https://vercel.com/docs/functions/edge-functions) that uses Prisma Client for talking to a database.
+
+</TopBlock>
+
+## General considerations when deploying to Vercel Edge Functions
+
+### Using an edge-compatible driver
+
+When deploying a Vercel Edge Function that uses Prisma ORM, you need to use an [edge-compatible driver](/orm/prisma-client/deployment/edge/overview#edge-compatibility-of-database-drivers) and its respective [driver adapter](/orm/overview/databases/database-drivers#driver-adapters) for Prisma ORM.
+
+> **Note**: [Prisma Accelerate](/accelerate) enables you to access _any_ database from _any_ edge function provider. No edge-compatible driver is necessary.
+
+The edge-compatible drivers for Vercel Edge Functions are:
+
+- [Neon Serverless](https://neon.tech/docs/serverless/serverless-driver) uses HTTP to access the database (also compatible with [Vercel Postgres](https://vercel.com/docs/storage/vercel-postgres))
+- [PlanetScale Serverless](https://planetscale.com/docs/tutorials/planetscale-serverless-driver) uses HTTP to access the database
+- [`@libsql/client`](https://github.com/tursodatabase/libsql-client-ts) is used to accessed Turso databases
+
+> **Note**: There's [also work being done](https://github.com/sidorares/node-mysql2/pull/2289) on the `node-mysql2` driver which will enable to access traditional MySQL databases from edge functions as well.
+
+### Setting your database connection URL as an environment variable
+
+First, ensure that the `DATABASE_URL` is set as the `url` of the `datasource` in your Prisma schema:
```

Comment:
Why is this relevant?
The `url` value from the Prisma schema is not actually used by the edge compatible Prisma Client.

Reply:
Not sure I understand. All the instructions in the EA docs set the `url` of the `datasource` to this env var (e.g. [here](https://www.notion.so/prismaio/Prisma-Vercel-Edge-Functions-and-Middleware-9e6f2a3bdb2a409caa9f8e4dba9e9bae?pvs=4#638f54c29dbf44798c9825405d14e10b)):

```prisma
datasource db {
  provider = "postgresql"
  url      = env("DATABASE_URL")
}
```

If Prisma Client doesn't get the DB URL from the Prisma schema, where does it get it from?

Reply:
Yes, exactly. That is the "weird" bit about driver adapters and should be explained to users.

Reply:
agree, I don't think this is documented anywhere yet! I just added a new section to the driver adapters page (currently deploying):

<img width="473" alt="image" src="https://github.com/prisma/docs/assets/4058327/2a0ed9a4-9842-4da1-94ab-2b01e7920420">